src/components/EditorIcon.tsx

```diff
@@ -1,36 +1,54 @@
 import React from "react";
-import PropTypes from "prop-types";
 import { withStyles } from "../themes";
+import { EditorInterface } from '../editors/editor';
 
-const EditorIcon = ({ enabled, img, name, styles, css }) => {
-  return (
-    <div {...css(styles.wrapper)}>
-      <img
-        {...css(styles.img, enabled ? styles.enabled : styles.disabled)}
-        alt={name}
-        src={img}
-      />
-    </div>
-  );
-};
+interface IProps {
+  enabled: boolean;
+  img: string;
+  name: string;
+  height: number | string;
+  width: number | string;
+  maxHeight: number | string;
+  styles: IStyles;
+  instance: EditorInterface;
+  css: Function;
+}
+interface IState {
+  enabled: boolean;
+}
+interface IStyles {
+  wrapper: Object;
+  img: Object;
+  enabled: Object;
+  disabled: Object;
+}
+class EditorIcon extends React.Component<IProps, IState> {
```

Comment:
Let's use `PureComponent` wherever we can, so this component should be stateless.